### PR TITLE
ONC 7 set booster cards

### PR DESCRIPTION
**Pull request by Simisays**
Should be good

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ManaCost:6 R R
+Types:Legendary Creature Dragon
+PT:5/4
+K:Affinity:Artifact
+K:Flying
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigExile | TriggerDescription$ Whenever CARDNAME attacks, exile the top five cards of your library. You may cast an artifact spell from among them this turn. If you do, it has affinity for artifacts.
+SVar:TrigExile:DB$ Dig | Defined$ You | DigNum$ 5 | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | StaticAbilities$ STPlay | Triggers$ Play1 | RememberObjects$ Remembered | ForgetOnMoved$ Exile | SubAbility$ DBCleanup
+SVar:STPlay:Mode$ Continuous | EffectZone$ Command | AffectedZone$ Exile,Stack | AddKeyword$ Affinity:Artifact | Affected$ Card.IsRemembered+Artifact | MayPlay$ True | Description$ You may play one of those cards until end of turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt:11]: play / cast aren't the same
(it doesn't allow _Ancient Den_)

File: forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+SVar:Play1:Mode$ SpellCast | ValidCard$ Card.IsRemembered | ValidActivatingPlayer$ You | TriggerZones$ Command | Execute$ ExileSelf | Static$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt:12]: Where did you copy this from?
ForgetOnMoved should already include it

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt:12]: Jodah, the Unifier

File: forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt]: I wonder would it be better to use ExiledwithSource here or is this ok ?

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt]: Imo the Affinity results only from this MayPlay Permission, meaning if you'd find another way to cast it should probably not apply

there are not many but you could use _Squee, the Immortal_

maybe the static can be split with `MayPlaySource`...?

File: forge-gui/res/cardsfolder/upcoming/lux_artillery.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Lux Artillery
+ManaCost:4
+Types:Artifact
+T:Mode$ SpellCast | ValidCard$ Artifact.Creature | ValidActivatingPlayer$ You | Execute$ TrigAnimate | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast an artifact creature spell, it gains sunburst. (It enters the battlefield with a +1/+1 counter on it for each color of mana spent to cast it.)
+SVar:TrigAnimate:DB$ Animate | Keywords$ Sunburst | Defined$ TriggeredCard | Duration$ Permanent
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ Player | ConditionCheckSVar$ X | ConditionCompare$ GE30 | Execute$ TrigDealDamage | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of your end step, if there are thirty or more counters among artifacts and creatures you control, CARDNAME deals 10 damage to each opponent.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/lux_artillery.txt:6]: Wrong ValidPlayer

File: forge-gui/res/cardsfolder/upcoming/lux_artillery.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+SVar:TrigDealDamage:DB$ DealDamage | Defined$ Opponent | NumDmg$ 10
+SVar:X:Count$Valid Artifact.YouCtrl+Other$CardCounters.P1P1
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/lux_artillery.txt:8]: Incomplete Valid

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/lux_artillery.txt:8]: Wrong CounterType

File: forge-gui/res/cardsfolder/upcoming/mirage_mockery.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Name:Mirage Mockery
+ManaCost:2 U
+Types:Sorcery
+K:Entwine:2 U
+A:SP$ Charm | Choices$ DBCopy1,DBCopy2 | CharmNum$ 1
+SVar:DBCopy1:DB$ CopyPermanent | ValidTgts$ Artifact.Creature+YouCtrl | TgtPrompt$ Select target artifact creature you control | SpellDescription$ Create a token that's a copy of target artifact creature you control
+SVar:DBCopy2:DB$ CopyPermanent | ValidTgts$ Creature.nonArtifact+YouCtrl | TgtPrompt$ Select target nonartifact creature you control | AddTypes$ Artifact | SpellDescription$ Create a token that's a copy of target nonartifact creature you control.
+DeckHas:Ability$Token
+Oracle:Choose one- \n* Create a token that's a copy of target artifact creature you control.\n*Create a token that's a copy of target nonartifact creature you control.\nEntwine {2}{U} (Choose both if you pay the entwine cost.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mirage_mockery.txt:9]: Bad formatting

File: forge-gui/res/cardsfolder/upcoming/mirage_mockery.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+Name:Mirage Mockery
+ManaCost:2 U
+Types:Sorcery
+K:Entwine:2 U
+A:SP$ Charm | Choices$ DBCopy1,DBCopy2 | CharmNum$ 1
+SVar:DBCopy1:DB$ CopyPermanent | ValidTgts$ Artifact.Creature+YouCtrl | TgtPrompt$ Select target artifact creature you control | SpellDescription$ Create a token that's a copy of target artifact creature you control
+SVar:DBCopy2:DB$ CopyPermanent | ValidTgts$ Creature.nonArtifact+YouCtrl | TgtPrompt$ Select target nonartifact creature you control | AddTypes$ Artifact | SpellDescription$ Create a token that's a copy of target nonartifact creature you control.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mirage_mockery.txt:7]: Why AddTypes$?

File: forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+Name:Synthesis Pod
+ManaCost:3 PU
+Types:Artifact
+A:AB$ DigUntil | Cost$ 1 PG T ExileFromStack<1/Spell.YouCtrl/spell you control> | ValidTgts$ Opponent | Valid$ Card.cmcEQX | FoundDestination$ Exile | RevealedDestination$ Library | RememberFound$ True | Shuffle$ True | SubAbility$ DBPlay | SpellDescription$ Target opponent reveals cards from the top of their library until they reveal a card with mana value equal to 1 plus the exiled spell's mana value. Exile that card, then that player shuffles. You may cast that exiled card without paying its mana cost.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt:4]: wrong cost

File: forge-gui/res/cardsfolder/upcoming/tangleweave_armor.txt [hunk 8]
```diff
@@ -0,0 +1,10 @@
+Name:Tangleweave Armor
+ManaCost:2 G G
+Types:Artifact Equipment
+K:Living Weapon
+K:Equip:4
+S:Mode$ Continuous | Affected$ Creature.EquippedBy | AddPower$ X | AddToughness$ X | Description$ Equipped creature gets +X/+X, where X is the greatest mana value among your commanders.
+SVar:X:Count$ValidBattlefield,Command,Stack,Graveyard,Exile,Hand,Library Card.IsCommander+YouOwn$GreatestCMC
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tangleweave_armor.txt:7]: you should shorten it with ValidAll

File: forge-gui/res/cardsfolder/upcoming/lux_artillery.txt [hunk 9]
```diff
@@ -0,0 +1,11 @@
+Name:Lux Artillery
+ManaCost:4
+Types:Artifact
+T:Mode$ SpellCast | ValidCard$ Artifact.Creature | ValidActivatingPlayer$ You | Execute$ TrigAnimate | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast an artifact creature spell, it gains sunburst. (It enters the battlefield with a +1/+1 counter on it for each color of mana spent to cast it.)
+SVar:TrigAnimate:DB$ Animate | Keywords$ Sunburst | Defined$ TriggeredCard | Duration$ Permanent
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | ConditionCheckSVar$ X | ConditionCompare$ GE30 | Execute$ TrigDealDamage | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of your end step, if there are thirty or more counters among artifacts and creatures you control, CARDNAME deals 10 damage to each opponent.
+SVar:TrigDealDamage:DB$ DealDamage | Defined$ Opponent | NumDmg$ 10
+SVar:X:Count$Valid Artifact.YouCtrl,Creature.YouCtrl$CardCounters
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/lux_artillery.txt:8]: you need to test that
I think it could either crash or justnot work

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/lux_artillery.txt]: I don't think it worked if you didn't use "ALL"

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/lux_artillery.txt]: Yes, I have tested it with All and it seems to work now

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/lux_artillery.txt]: Doesn't seem correct to me :p

grafik

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/lux_artillery.txt]: Oh I see tested it with thirty counters on board

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/lux_artillery.txt]: Turns out All needs to be ALL

File: forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt]: Hmm it seems like Forge handles this situation correctly 
image
image

File: forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigExile | TriggerDescription$ Whenever CARDNAME attacks, exile the top five cards of your library. You may cast an artifact spell from among them this turn. If you do, it has affinity for artifacts.
+SVar:TrigExile:DB$ Dig | Defined$ You | DigNum$ 5 | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | StaticAbilities$ STPlay | RememberObjects$ Remembered | ForgetOnMoved$ Exile | SubAbility$ DBCleanup
+SVar:STPlay:Mode$ Continuous | EffectZone$ Command | AffectedZone$ Exile,Stack | AddKeyword$ Affinity:Artifact | Affected$ Card.IsRemembered+Artifact+nonLand | MayPlay$ True | Description$ You may play one of those cards until end of turn.
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:HasAttackEffect:TRUE
+DeckNeeds:Type$Artifact
+Oracle:Affinity for artifacts (This spell costs {1} less to cast for each artifact you control.)Flying, haste\nWhenever Chiss-Goria, Forge Tyrant attacks, exile the top five cards of your library. You may cast an artifact spell from among them this turn. If you do, it has affinity for artifacts.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt:15]: missing \n

**Comment by Simisays** [comment]
Hmm Synthetic Pod doesn't allow me to target permanents in Stack. Is this an engine problem or does my script need some more work ?

File: forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt [hunk 11]
```diff
@@ -0,0 +1,9 @@
+Name:Synthesis Pod
+ManaCost:3 PU
+Types:Artifact
+A:AB$ DigUntil | Cost$ 1 UP T ExileFromStack<1/Spell.YouCtrl/spell you control> | ValidTgts$ Opponent | Valid$ Card.cmcEQX | FoundDestination$ Exile | RevealedDestination$ Library | ImprintFound$ True | Shuffle$ True | SubAbility$ DBPlay | SpellDescription$ Target opponent reveals cards from the top of their library until they reveal a card with mana value equal to 1 plus the exiled spell's mana value. Exile that card, then that player shuffles. You may cast that exiled card without paying its mana cost.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt:4]: You can probably use `Card.YouCtrl` for now

File: forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt]: Ideally yes but for now missing support for timestamp and in ForgetOnMoved make that too unreliable

File: forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt (hunk 11 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt:4]: Yes that made it work, but after further testing it seems that SVar:X:Exiled$CardManaCost/Plus.1 isn't working at all. I guess CostExileFromStack.java needs some additions ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt:4]: You can try this in payAsDecided:
`ability.addCostToHashList(CardUtil.getLKICopy(sa.getHostCard()), "Exiled");`

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt:4]: That made it work ! Thanks for the help as usual 
image

File: forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt]: Fixed

File: forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt (hunk 11 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/synthesis_pod.txt]: You should create the LKI as the first thing, not after exiling

**Review by tool4ever — APPROVED** [review]

**Pull request merged**